README.md

```diff
@@ -1,22 +1,70 @@
+
+## Installation and Testing
+
+First, ensure that, Node.js and Yarn have been already installed:
 
 ```bash
-# 1. Installs dependencies
-yarn install
+$ git clone https://github.com/synapseweb3/ibc-solidity-contract
+$ cd ibc-solidity-contract
+$ yarn install
+$ yarn compile
+$ yarn test
+```
+
+## Deploy on Axon
 
-# 2. Compile
-yarn compile
+Creating a `.env` file in the project root path:
+
+```
+AXON_HTTP_RPC_URL=http://<AXON_URL>:<HTTP_PORT>
+AXON_WS_RPC_URL=ws://<AXON_URL>:<WS_PORT>
+```
+
+Then, use the migration command to deploy the contract and prepare basic settings for Axon:
+
+```bash
+$ yarn migrate
 ```
 
-## Deployment on AXON
+After deployment, each standalone component in the project has been migrated into Axon and the entry is `OwnableIBCHandler` contract, which accepts operations, including registering IBC module and binding channel port.
 
-Create `.env` in project root and put axon url like
+## Register Light Client
 
+Registering light client is essential for verifying transactions from the counterparty chain. A mock light client has been registered into OwnableIBCHandler contract during the migration, which acts as an entry for counterparty.
+
+While the official light clients for verifying Cosmos and CKB transactions are in development, you can use a custom light client by following the [method](https://github.com/synapseweb3/ibc-solidity-contract/blob/master/contracts/core/02-client/ILightClient.sol) below 
+
+```solidity
+function registerClient(string calldata clientType, ILightClient client)
+```
+
+Note: clientType here can be one of these values: “07-axon”, “07-tendermint”, or “07-ckb4ibc”.
+
+## Bind IBC Module
+
+Before binding the IBC Module, please follow the [MockModule](https://github.com/synapseweb3/ibc-solidity-contract/blob/master/contracts/apps/20-transfer/MockModule.sol) example to create your custom business module contract. Once you've deployed this contract in Axon, you can register its address within the `OwnableIBCHandler` contract by binding it to a channel port:
+
+```solidity
+function bindPort(string calldata portId, address moduleAddress)
 ```
-AXON_HTTP_RPC_URL=http://<Ip>:<Port>
+
+To comply with IBC protocol regulations, a channel cannot function until it is paired with a business module through a port.
+
+## Send ICS20 Packet
+
+In the previous migration step, a standard ICS20 transfer module has already been bound at port `port-0` by default. Run script `scripts/send_packet.js` to send a standard custom ICS20 packet:
+
+```bash
+$ yarn send
```

Comment:
❯ yarn send
yarn run v1.22.19
$ HTTPS_PROXY= HTTP_PROXY= truffle exec './scripts/send_packet.js' --network axon
Using network 'axon'.
AXON_HTTP_RPC_URL=https://rpc-alphanet-axon.ckbapp.dev

send packet from 07-axon-0 to 07-axon-1
TypeError: Cannot set properties of undefined (setting 'onopen')
Truffle v5.11.1 (core: 5.11.1)
Node v18.17.1
error Command failed with exit code 1.

Reply:
`send` is unworkable because I just use it for debug use, but it's weird if it doesn't show in `README.md`, so I wrote it and consider to complete the `send` command later.

Reply:
LGTM